src/server/scripts/Northrend/Naxxramas/naxxramas.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "InstanceScript.h"
+#include "Map.h"
+#include "MotionMaster.h"
+#include "Player.h"
+#include "ScriptedCreature.h"
+#include "TemporarySummon.h"
+#include "naxxramas.h"
+
+enum NaxxEvents
+{
+    EVENT_SUMMON_FROGGER_WAVE = 1
+};
+
+struct npc_frogger_trigger_naxx : public ScriptedAI
```

Comment:
Is there a specific reason why this can't be in `instance_naxxramas.cpp`?

Reply:
I've seen other instance models and I think it should be separate, isn't it correct?

Reply:
Hmm, indeed. That has slipped my attention.

Alright, carry on ;)